### PR TITLE
Add proposal requirements and template

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

- Created proposal template at `docs/proposal-000-template.md` (name up for debate).
- Updated Proposal section in Contributing doc
- Added k8s' toc generator to build deps for easy tocing (Do we actually add stuff like this to build deps tho? it's not _technically_ required, just handy. LMK before I rebuild any images.)

This is all up for debate, please disagree with any headings or ask for more. This is a mashup of Kep, Mark E's ideas and mine. Hopefully we can settle on a standard which will work for us.

No rush, merged by EOD 29th would be neat if possible.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by Callisto13** [comment]
let's take the rebase action for a test drive

**Comment by Callisto13** [comment]
oh snap

**Review by aclevername — APPROVED** [review]
LGTM

**Review by michaelbeaumont — APPROVED** [review]

File: .requirements [hunk 1]
```diff
@@ -13,3 +13,4 @@ k8s.io/code-generator/cmd/deepcopy-gen
 k8s.io/code-generator/cmd/defaulter-gen
 k8s.io/code-generator/cmd/informer-gen
 k8s.io/code-generator/cmd/lister-gen
+sigs.k8s.io/mdtoc
```
Comment (michaelbeaumont) [inline .requirements:16]: probably not ideal here since it ends up in the build image but i don't think we have a better place to put it

Comment (Callisto13) [reply .requirements:16]: okay 😞

**Review by cPu1 — APPROVED** [review]
:tada:

**Pull request merged**